compose/config/validation.py

```diff
@@ -44,6 +48,20 @@ def format_ports(instance):
     return True
 
 
+@FormatChecker.cls_checks(
+    format="environment")
+def format_boolean_in_environment(instance):
+    """
+    Check if there is a boolean in the environment and display a warning.
+    Always return True here so the validation won't raise an error.
+    """
+    if isinstance(instance, bool):
+        log.warn("Warning: There is a boolean value in the 'environment' key,")
+        log.warn("this will be deprecated in the future.")
+        log.warn("Please wrap booleans in quotes to ensure they're preserved. \r\n")
```

Comment:
I think we could include a bit more info in this error message, maybe something like:

```python
log.warn("Warning: There is a boolean value '{inst}' in the 'environment' key. Environment variables "
         "can only be strings. Please add quotes to any boolean values to make them strings"
         " (ex: '{inst}'). This warning will become an error in a future release.".format(inst=instance))
```